test/test_type_promotion.py

```diff
@@ -225,6 +225,56 @@ def test_result_type(self):
         self.assertEqual(torch.result_type(torch.tensor([1., 1.], dtype=torch.float), 1.), torch.float)
         self.assertEqual(torch.result_type(torch.tensor(1., dtype=torch.float), torch.tensor(1, dtype=torch.double)), torch.double)
 
+    def test_sparse(self):
+        t = torch.ones([5, 5], dtype=torch.int, device=self.device) * 5
+        t[0, 0] = 0
+        t[1, 1] = 0
+        t[2, 2] = 0
+        s = t.to_sparse()
+        self.assertEqual((s + s).to_dense(), t + t)
+        self.assertEqual(t + s, (s + s).to_dense())
+        self.assertEqual(t + s.to(torch.int), (s + s).to_dense())
+        self.assertEqual(t + s.to(torch.double), (t + t).to(torch.double))
```

Comment:
does assertEqual check the dtype?  I know some version did/didn't.

Reply:
it doesn't. I'll add a few more explicit dtype checks.